src/basic/macro.h

```diff
@@ -323,6 +323,8 @@ static inline int __coverity_check_and_return__(int condition) {
         ({                                              \
                 typeof(x) _x_ = (x);                    \
                 size_t ans = 1;                         \
+                if (IS_SIGNED_INTEGER_TYPE(_x_))        \
+                        ans++;                          \
```

Comment:
Not a big matter, but is it possible to do like the following?
```
                if (IS_SIGNED_INTEGER_TYPE(_x_))        \
                        ans += _x_ < 0;
```

Reply:
Unfortunately, that still makes gcc complain:

```
[443/1827] Compiling C object src/shared/libsystemd-shared-250.a.p/json.c.o
In file included from ../src/basic/alloc-util.h:9,
                 from ../src/shared/json.c:12:
../src/shared/json.c: In function ‘print_source’:
../src/basic/macro.h:327:36: warning: comparison of unsigned expression in ‘< 0’ is always false [-Wtype-limits]
  327 |                         ans += _x_ < 0;                 \
      |                                    ^
../src/shared/json.c:1466:17: note: in expansion of macro ‘DECIMAL_STR_WIDTH’
 1466 |                 DECIMAL_STR_WIDTH(v->source->max_line) :
      |  
```

Reply:
I also tried to use `DISABLE_WARNING_TYPE_LIMITS` and `REENABLE_WARNING` around the condition, but that just triggers a different compilation issue.

Reply:
OK. Let's keep this as is. One extra buffer space should not cause anything serious.

Reply:
`ans += __builtin_choose_expr(IS_SIGNED_INTEGER_TYPE(_x_), 1, 0)`; should do the trick

Reply:
Indeed, thanks for the hint!

Reply:
> `ans += __builtin_choose_expr(IS_SIGNED_INTEGER_TYPE(_x_), 1, 0)`; should do the trick

Hm? What is the merit to use `__builtin_choose_expr()`? Why not to use the following?
```
size_t ans = 1 + IS_SIGNED_INTEGER_TYPE(_x_);
```

Reply:
You have a point. Though, `IS_SIGNED_INTEGER_TYPE(_x_) ? 2 : 1` is preferred by coding style.